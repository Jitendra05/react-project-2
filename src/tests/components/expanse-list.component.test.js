import React from 'react';
import {shallow} from 'enzyme';
import {ExpanseList} from '../../components/expanse-list.component';
import {expanses} from '../fixtures/expanses-data';

test('should render ExpanseList component correctly with expanses',()=>{
    const wrapper = shallow(<ExpanseList expanses={expanses}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpanseList component correctly with no expanses',()=>{
    const wrapper = shallow(<ExpanseList expanses={[]}/>);
    expect(wrapper).toMatchSnapshot();
});


