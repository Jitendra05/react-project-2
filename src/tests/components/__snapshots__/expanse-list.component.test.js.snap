// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpanseList component correctly with expanses 1`] = `
<div>
  <[object Object]
    expanses={
      Array [
        Object {
          "amount": 9000,
          "createdAt": -4000,
          "description": "rent",
          "id": "1",
          "note": "",
        },
        Object {
          "amount": 900,
          "createdAt": 1000,
          "description": "gas bill",
          "id": "2",
          "note": "",
        },
        Object {
          "amount": 1200,
          "createdAt": 4000,
          "description": "water bill",
          "id": "3",
          "note": "",
        },
      ]
    }
  />
</div>
`;

exports[`should render ExpanseList component correctly with no expanses 1`] = `
<div>
  <div
    className="container"
  >
    <h3
      className="table-heading"
    >
      Please add your expenses.
    </h3>
  </div>
</div>
`;
